### PR TITLE
Rename multiple nRF-based boards

### PR DESCRIPTION
**Pull request by carlescufi**
Renames the following:

nrf52_pca20020: Rename to thingy52_nrf52832
nrf52840_pca10059: Rename to nrf52840dongle_nrf52840
nrf52833_pca10100: Rename to nrf52833dk_nrf52833
nrf52810_pca10040: Rename to nrf52dk_nrf52810
nrf52_pca10040: Rename to nrf52dk_nrf52832
nrf51_pca10031: Rename to nrf51dongle_nrf51422

### CODE REVIEW

**Review by anangl — COMMENTED** [review]
A few missed renamings for nrf52810_pca10040:
```
boards/arm/nrf52dk_nrf52810/Kconfig.defconfig:1:# nRF52810 PCA10040 board configuration
boards/arm/nrf52dk_nrf52810/Kconfig.board:1:# nRF52810 PCA10040 board configuration
tests/kernel/sched/metairq/testcase.yaml:5:    platform_exclude: nrf52810_pca10040
tests/kernel/sched/preempt/testcase.yaml:5:    platform_exclude: nrf52810_pca10040
tests/kernel/poll/testcase.yaml:6:    platform_exclude: nrf52810_pca10040
tests/bluetooth/shell/testcase.yaml:6:    platform_exclude: nrf52810_pca10040
tests/bluetooth/shell/testcase.yaml:15:    platform_exclude: nrf52810_pca10040
tests/bluetooth/mesh_shell/testcase.yaml:4:    platform_exclude: nrf52810_pca10040
subsys/bluetooth/controller/Kconfig:973:	depends on BOARD_NRF51DK_NRF51422 || BOARD_NRF52DK_NRF52832 || BOARD_NRF52810_PCA10040 || BOARD_NRF52840DK_NRF52840 || BOARD_RV32M1_VEGA
subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/debug.h:22:	defined(CONFIG_BOARD_NRF52810_PCA10040)
```
and unwanted occurrences of PCA10040:
```
boards/arm/nrf52dk_nrf52832/doc/index.rst:31:     :alt: nRF52 PCA10040 DK
boards/arm/nrf52dk_nrf52832/doc/index.rst:33:     nRF52 PCA10040 DK (Credit: Nordic Semiconductor)
```
Also a typo (nrf2833dk) in a few file names:
```
./boards/arm/nrf52833dk_nrf52833/nrf2833dk_nrf52833.yaml
./boards/arm/nrf52833dk_nrf52833/nrf2833dk_nrf52833_defconfig
./boards/arm/nrf52833dk_nrf52833/nrf2833dk_nrf52833.dts
./tests/drivers/counter/counter_basic_api/boards/nrf2833dk_nrf52833.overlay
./tests/drivers/counter/counter_basic_api/boards/nrf2833dk_nrf52833.conf
```

File: doc/conf.py [hunk 1]
```diff
@@ -345,11 +345,11 @@
 		('boards/arm/nrf52840_pca10056/doc/nrf52840_pca10056/index', 'boards/arm/nrf52840dk_nrf52840/doc/index'),
 		('boards/arm/nrf9160_pca10090/doc/nrf9160_pca10090/index', 'boards/arm/nrf9160dk_nrf9160/doc/index'),
 		('boards/arm/nrf52840_pca10090/doc/nrf52840_pca10090/index', 'boards/arm/nrf9160dk_nrf52840/doc/index'),
-		('boards/arm/nrf52840_pca10059/doc/nrf52840_pca10059', 'boards/arm/nrf52840_pca10059/doc/index'),
+		('boards/arm/nrf52840_pca10059/doc/nrf52840_pca10059', 'boards/arm/nrf52840dongle_nrf52840/doc/index'),
```
Comment (anangl) [inline doc/conf.py:348]: You need to add an additional entry for each one you modify here, see https://github.com/zephyrproject-rtos/zephyr/pull/23447#discussion_r396496580.

**Comment by nvlsianpu** [comment]
Please point https://github.com/zephyrproject-rtos/mcuboot/pull/11 in the manifest for make CI more happy.

File: boards/arm/nrf51dk_nrf51422/doc/index.rst [hunk 2]
```diff
@@ -6,7 +6,7 @@ nRF51 DK
 Overview
 ********
 
-The nRF51 Development Kit (nRF51 DK) hardware provides support for the Nordic
+The nRF51 Development Kit (PCA10028) hardware provides support for the Nordic
```
Comment (ioannisg) [inline boards/arm/nrf51dk_nrf51422/doc/index.rst:9]: nice

File: boards/arm/nrf51dongle_nrf51422/doc/index.rst [hunk 3]
```diff
@@ -1,7 +1,7 @@
-.. _nrf51_pca10031:
+.. _nrf51dongle_nrf51422:
 
-nRF51-PCA10031
-##############
+nRF51-Dongle
```
Comment (ioannisg) [inline boards/arm/nrf51dongle_nrf51422/doc/index.rst:3]: "-" is not really needed here (we don't have this in the other Doc titles)

**Review by ioannisg — COMMENTED** [review]
you need to add the notes in the .rst documentation about how the boards were called in previous releases - see how this is done for 91.

File: boards/deprecated.cmake [hunk 4]
```diff
@@ -9,7 +9,13 @@
 # set(<old_board_name>_DEPRECATED <new_board_name>)
 
 set(nrf51_pca10028_DEPRECATED nrf51dk_nrf51422)
+set(nrf51_pca10031_DEPRECATED nrf51dongle_nrf51422)
+set(nrf52_pca10040_DEPRECATED nrf52dk_nrf52832)
+set(nrf52810_pca10040_DEPRECATED nrf52dk_nrf52810)
+set(nrf52833_pca10100_DEPRECATED nrf52833dk_nrf52833)
+set(nrf52_pca20020_DEPRECATED thingy52_nrf52832)
```
[the remark this answers is not in the document]
Comment (carlescufi) [reply boards/deprecated.cmake:16]: But the are not in alphabetical order now? 52811 comes after 52840?

File: doc/conf.py (hunk 1 above)
Comment (ioannisg) [reply doc/conf.py:348]: yes :)

File: boards/deprecated.cmake (hunk 4 above)
[the remark this answers is not in the document]
Comment (anangl) [reply boards/deprecated.cmake:16]: Yes, I know. I introduced this disorder myself... Thus the word "keep" was not a best choice for my suggestion, I should have used "make" instead.

**Comment by carlescufi** [comment]
> Please point [zephyrproject-rtos/mcuboot#11](https://github.com/zephyrproject-rtos/mcuboot/pull/11) in the manifest for make CI more happy.

Thank you, done!

File: boards/arm/nrf52840dongle_nrf52840/nrf52840dongle_nrf52840.dts [hunk 5]
```diff
@@ -9,8 +9,8 @@
 #include <nordic/nrf52840_qiaa.dtsi>
 
 / {
-	model = "Nordic PCA10059 Dev Kit";
-	compatible = "nordic,pca10059-dk", "nordic,nrf52840-qiaa",
+	model = "Nordic nRF52840 Dongle NRF52840 Dev Kit";
```
[the remark this answers is not in the document]
Comment (carlescufi) [reply boards/arm/nrf52840dongle_nrf52840/nrf52840dongle_nrf52840.dts]: Done

File: boards/arm/nrf52840dongle_nrf52840/Kconfig.board [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# nRF52840 Dongle NRF52840 board configuration
+
+# Copyright (c) 2018 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_NRF52840DONGLE_NRF52840
+	bool "NRF52840 DONGLE NRF52840"
```
[the remark this answers is not in the document]
Comment (carlescufi) [reply boards/arm/nrf52840dongle_nrf52840/Kconfig.board]: Done

File: boards/arm/nrf52dk_nrf52810/nrf52dk_nrf52810.dts [hunk 7]
```diff
@@ -10,8 +10,8 @@
 #include <nordic/nrf52810_qfaa.dtsi>
 
 / {
-	model = "Nordic PCA10040 Dev Kit";
-	compatible = "nordic,pca10040-dk", "nordic,nrf52810-qfaa",
+	model = "Nordic nRF52 DK NRF52810 Dev Kit";
```
[the remark this answers is not in the document]
Comment (carlescufi) [reply boards/arm/nrf52dk_nrf52810/nrf52dk_nrf52810.dts]: Done

File: boards/arm/nrf52dk_nrf52832/Kconfig.board [hunk 8]
```diff
@@ -1,8 +1,8 @@
-# nRF52 PCA10040 board configuration
+# nRF52 DK board configuration
 
 # Copyright (c) 2016 Nordic Semiconductor ASA
 # SPDX-License-Identifier: Apache-2.0
 
-config BOARD_NRF52_PCA10040
-	bool "nRF52 PCA10040"
+config BOARD_NRF52DK_NRF52832
+	bool "nRF52 DK"
```
[the remark this answers is not in the document]
Comment (carlescufi) [reply boards/arm/nrf52dk_nrf52832/Kconfig.board]: Done

File: boards/arm/nrf52dk_nrf52832/Kconfig.board [hunk 9]
```diff
@@ -1,8 +1,8 @@
-# nRF52 PCA10040 board configuration
+# nRF52 DK board configuration
```
[the remark this answers is not in the document]
Comment (carlescufi) [reply boards/arm/nrf52dk_nrf52832/Kconfig.board]: Done

File: boards/arm/nrf52dk_nrf52832/Kconfig.defconfig [hunk 10]
```diff
@@ -1,12 +1,12 @@
-# nRF52 PCA10040 board configuration
+# nRF52 DK board configuration
```
[the remark this answers is not in the document]
Comment (carlescufi) [reply boards/arm/nrf52dk_nrf52832/Kconfig.defconfig]: Done

File: boards/arm/nrf52dk_nrf52832/nrf52dk_nrf52832.yaml [hunk 11]
```diff
@@ -1,5 +1,5 @@
-identifier: nrf52_pca10040
-name: nRF52-PCA10040
+identifier: nrf52dk_nrf52832
+name: nRF52-DK
```
[the remark this answers is not in the document]
Comment (carlescufi) [reply boards/arm/nrf52dk_nrf52832/nrf52dk_nrf52832.yaml]: Done

File: boards/arm/thingy52_nrf52832/Kconfig.board [hunk 12]
```diff
@@ -1,9 +1,9 @@
-# nRF52 PCA20020 board configuration
+# Thingy52 NRF52832 board configuration
 
 # Copyright (c) 2018 Aapo Vienamo
 # SPDX-License-Identifier: Apache-2.0
 
-config BOARD_NRF52_PCA20020
-	bool "nRF52 PCA20020"
+config BOARD_THINGY52_NRF52832
+	bool "THINGY52 NRF52832"
```
[the remark this answers is not in the document]
Comment (carlescufi) [reply boards/arm/thingy52_nrf52832/Kconfig.board:7]: Done

File: doc/conf.py (hunk 1 above)
Comment (carlescufi) [reply doc/conf.py:349]: Done

File: boards/arm/nrf51dongle_nrf51422/doc/index.rst (hunk 3 above)
Comment (carlescufi) [reply boards/arm/nrf51dongle_nrf51422/doc/index.rst:3]: Done

**Comment by carlescufi** [comment]
> you need to add the notes in the .rst documentation about how the boards were called in previous releases - see how this is done for 91.

Thank you, done.

**Comment by carlescufi** [comment]
> A few missed renamings for nrf52810_pca10040:
> 
> ```
> boards/arm/nrf52dk_nrf52810/Kconfig.defconfig:1:# nRF52810 PCA10040 board configuration
> boards/arm/nrf52dk_nrf52810/Kconfig.board:1:# nRF52810 PCA10040 board configuration
> tests/kernel/sched/metairq/testcase.yaml:5:    platform_exclude: nrf52810_pca10040
> tests/kernel/sched/preempt/testcase.yaml:5:    platform_exclude: nrf52810_pca10040
> tests/kernel/poll/testcase.yaml:6:    platform_exclude: nrf52810_pca10040
> tests/bluetooth/shell/testcase.yaml:6:    platform_exclude: nrf52810_pca10040
> tests/bluetooth/shell/testcase.yaml:15:    platform_exclude: nrf52810_pca10040
> tests/bluetooth/mesh_shell/testcase.yaml:4:    platform_exclude: nrf52810_pca10040
> subsys/bluetooth/controller/Kconfig:973:	depends on BOARD_NRF51DK_NRF51422 || BOARD_NRF52DK_NRF52832 || BOARD_NRF52810_PCA10040 || BOARD_NRF52840DK_NRF52840 || BOARD_RV32M1_VEGA
> subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/debug.h:22:	defined(CONFIG_BOARD_NRF52810_PCA10040)
> ```
> 
> and unwanted occurrences of PCA10040:
> 
> ```
> boards/arm/nrf52dk_nrf52832/doc/index.rst:31:     :alt: nRF52 PCA10040 DK
> boards/arm/nrf52dk_nrf52832/doc/index.rst:33:     nRF52 PCA10040 DK (Credit: Nordic Semiconductor)
> ```
> 
> Also a typo (nrf2833dk) in a few file names:
> 
> ```
> ./boards/arm/nrf52833dk_nrf52833/nrf2833dk_nrf52833.yaml
> ./boards/arm/nrf52833dk_nrf52833/nrf2833dk_nrf52833_defconfig
> ./boards/arm/nrf52833dk_nrf52833/nrf2833dk_nrf52833.dts
> ./tests/drivers/counter/counter_basic_api/boards/nrf2833dk_nrf52833.overlay
> ./tests/drivers/counter/counter_basic_api/boards/nrf2833dk_nrf52833.conf
> ```

Hey, a big thank you for spotting all those issues.

**Review by ioannisg — APPROVED** [review]
Great work, Carles, thanks.

**Comment by anangl** [comment]
I added the DNM label so that this PR is not merged until the mcuboot revision is updated.

**Review by anangl — APPROVED** [review]

**Comment by carlescufi** [comment]
@nvlsianpu will you merge the MCUboot patch?

**Comment by nvlsianpu** [comment]
^^ sure

**Pull request merged**